Refactor Profile Page props to work in three possible modes (#24)
ValentinnDimitroff/ra-auth-ui: README.md

yarn run start-demo
```

## Available Props

| Name        | Type              | Default   | Description                                                                                             |
| ----------- | ----------------- | --------- | ------------------------------------------------------------------------------------------------------- |
| `authRoutes`  | array             | array     | *If unset provides default authentication routes and screens. But you can pass custom routes like `{ path: string; Component: FC }`*                                          |

> | `authLayout`  | object            | undefined | *If set to '{userMenu: true}' provides default UserMenu. If set to object pass on the props to Layout.* |

Reviewer: The prop is called `authOptions`.  It should point to the details of how the prop works, we can't explain all the features in the box here.

```suggestion
| `authOptions`  | object            | undefined | *If unset provides default Profile Page, integrated User Menu and password rules out of the box. All supported fields explained [below](#functionality-of-authlayout)* |
```

Author: Fair, will update accordingly.

```
Example
``` -->

> ## Functionality of authLayout

Reviewer: ```suggestion
## Functionality of authOptions
```

 Provides authenticated layout with smart defaults.

- ### Defaults 
    The default props can be set by `authOptions`

>     | Props         | Type    | Defaults    | Behavior                                                                                                                           |

Reviewer: As a rule of thumb we can copy the react-admin style of describing props: https://marmelab.com/react-admin/Admin.html#props

```suggestion
    | Props         | Type    | Defaults    | Description                                                                                                                           |
```

    | ------------- | ------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------- |

>     | profilePage   | boolean | true        | *By default profile page is enabled and a basic profile component is set.*                                                         |

Reviewer: ```suggestion
    | profilePage   | boolean | Component | true        | *By default profile page is enabled and a basic profile component is set. You can opt out by setting it to `false` or pass your own component which will be rendered at the `/profile` route*                                                         |
```

>     | userMenuItems | array   | empty array | *Not specified.*                                                                                                                   |

Reviewer: ```suggestion
    | userMenuItems | array   | empty array | *Populates the user menu with items depending on the `authOptions` configuration*                                                                                                                   |
```

>     | passwordRules | object  | all enabled | *By default all rules are enabled. A digit, lowercase, non-alphanumeric,uppercase,min-length is `8` and min one symbol is needed.* |

Reviewer: `loginRedirectPath` is missing in the table

- ### Supports object

  -   userMenu - if no `appBar` is passed
      -   true - default menu with profile link and logout button
      -   array of object - {to, ...}
      -   array of elements - TODO fnc receiving permissions and return element

  - Other supported keys
      -   menu
      -   appBar

>       -   sideBar

Reviewer: This is outdated

Author: removed

- ### profilePage - works in three possible modes:

  - `true` - enabled by default. Loads the default Profile component
  - `false` - removes the component and the route `/profile`
  - `custom component` - passes the custom component and renders it with the default route `/profile`

  *Note: The default routes can be changed*
  
- ### authOptions

  If unset provides default user menu ready to use. But you can pass custom profilePage, loginRedirectPath or array of userMenuItems to the user menu.

> - ### profilePage

Reviewer: This is duplicated and outdated
